tests/unit/dataframer/test_dataframer_smmart.py

```diff
@@ -0,0 +1,232 @@
+    resources = cursor.fetchall()
+    _resources = []
+    for resource in resources:
+        key, resource_type, resource = resource
+        resource = json.loads(resource)
+        _resources.append(resource)
+    return _resources
+
+
+def test_smmart_db(smmart_local_db, expected_keys):
+    """Simple test to verify db load."""
+    assert smmart_local_db
+    cursor = smmart_local_db.connection.cursor()
+    cursor.execute("SELECT * FROM resources")
+    resources = cursor.fetchall()
+    actual_keys = []
+    for resource in resources:
+        key, resource_type, resource = resource
+        _ = json.loads(resource)
+        actual_keys.append(key)
+    actual_keys = sorted(actual_keys)
+    print(actual_keys)
+    assert actual_keys == expected_keys
+
+
+def test_simplified(smmart_resources, simplified_smmart_resources):
+    """Simple test to verify resources are simplified (no joins)."""
+    actual = {}
+    for resource in smmart_resources:
+        assert isinstance(resource, dict), f"Expected dict, got {type(resource)} {resource}"
+        simplified = SimplifiedResource.build(resource=resource).simplified
+        actual[f"{simplified['resourceType']}/{simplified['id']}"] = simplified
+    print(actual)
+    assert actual == simplified_smmart_resources
+
+
+def test_smmart_document_reference(smmart_local_db, document_reference_dataframe):
+    """Test the dataframer using a local database with a SMMART bundle, this test ensures document reference and all its Observations."""
+    # TODO - once we are happy w/ this, move it to gen3_tracker.meta.dataframer
+    cursor = smmart_local_db.connection.cursor()
+
+    # get the document reference
+    document_reference_key = 'DocumentReference/9ae7e542-767f-4b03-a854-7ceed17152cb'
+    cursor.execute("SELECT * FROM resources WHERE key = ?", (document_reference_key,))
+    _ = cursor.fetchone()
+    assert _, f"{document_reference_key} not found"
+    key, resource_type, resource = _
+    resource = json.loads(resource)
+
+    # simplify it
+    simplified = SimplifiedResource.build(resource=resource).simplified
+    document_reference = resource
+
+    # get its subject, simplify it and add it to the simplified document reference
+    subject_key = document_reference['subject']['reference']
+    cursor.execute("SELECT * FROM resources WHERE key = ?", (subject_key,))
+    _ = cursor.fetchone()
+    assert _, f"{subject_key} not found"
+    key, resource_type, resource = _
+    resource = json.loads(resource)
+    simplified_subject = SimplifiedResource.build(resource=resource).simplified
+    prefix = simplified_subject['resourceType'].lower()
+    for k, v in simplified_subject.items():
+        if k == 'resourceType':
+            continue
+        simplified[f"{prefix}_{k}"] = v
+
+    # get its focus, simplify it and add it to the simplified document reference
+    if 'focus' in document_reference and len(document_reference['focus']) > 0:
```

Comment:
this is untested tested since our fixture has no DocumentReference.focus, do we still want to keep it in?

Reply:
https://github.com/ACED-IDP/gen3_util/blob/763b1f1a1f3aa3e35b60be910662983e99c6aef9/tests/unit/dataframer/test_dataframer.py#L78-L82